### ISSUE TITLE
Support nostr address format

### ISSUE THREAD

**Issue by jolestar** [rooch-network/rooch#2209]
After #1770, all Bitcoin addresses can generate a rooch address. The nostr address is the same as the bitcoin taproot address type [bip340](https://bips.xyz/340) [nips01](https://github.com/nostr-protocol/nips/blob/master/01.md)

- [x] #2211
- [x] #2233
- [ ] #2464 SDK support generates the rooch address via the nostr address and provides a function to allow the nostr app to sign rooch tx via the nostr private key.  

relative issue #406 #451

**Comment by feliciss** [comment]
I think I can look the first issue @2211.

**Comment by jolestar** [comment]
@feliciss, Can you continue the second issue? It just updates the `RoochOrBitcoinAddress::from_str`, for supports parsing nostr bech32 address -> Bitcoin Address -> RoochAddress.

**Comment by feliciss** [comment]
> @feliciss, Can you continue the second issue? It just updates the `RoochOrBitcoinAddress::from_str`, for supports parsing nostr bech32 address -> Bitcoin Address -> RoochAddress.

Okay.

**Comment by feliciss** [comment]
Submitted a new issue #2235.

**Comment by feliciss** [comment]
Is the third task related to TypeScript?

**Comment by feliciss** [comment]
> > Is the third task related to TypeScript?
> 
> yes

I think it might be handled by @wow-sven ?
